src/geeknn/ordination/_base.py

```diff
@@ -0,0 +1,282 @@
+
+    type: str
+    coordinates: list[Any]
+
+
+class Feature(BaseModel):
+    """Client-side proxy for ee.Feature object."""
+
+    type: str
+    geometry: Geometry
+    id: str
+    properties: dict[str, Any]
+
+    def get_property(self, property: str) -> Any:
+        return self.properties.get(property)
+
+    def get_properties(self, properties: list[str]) -> list[Any]:
+        return [self.get_property(x) for x in properties]
+
+
+class FeatureCollection(BaseModel):
+    """Client-side proxy for ee.FeatureCollection object."""
+
+    type: str
+    columns: dict[str, Any]
+    version: int
+    id: str
+    properties: dict[str, Any]
+    features: list[Feature]
+
+    def aggregate_array(self, property: str) -> NDArray:
+        """Return a single property as a 1-D numpy array."""
+        return np.array([f.get_property(property) for f in self.features])
+
+    def properties_to_array(self, properties: list[str]) -> NDArray:
+        """Return specified properties as a 2-D numpy array."""
+        return np.array([f.get_properties(properties) for f in self.features])
+
+    @classmethod
+    def from_ee_feature_collection(
+        cls, fc: ee.FeatureCollection, num_threads: int = -1, chunk_size: int = 5000
+    ):
+        """Create a client-side FeatureCollection from an ee.FeatureCollection
+        using multiple threads and chunking to avoid memory issues.
+        """
+        size = fc.size().getInfo()
+        info = fc.limit(0).getInfo()
+        chunks = [fc.toList(chunk_size, i) for i in range(0, size, chunk_size)]
+
+        with Parallel(n_jobs=num_threads, backend="threading") as p:
+            chunk_data = p(delayed(chunk.getInfo)() for chunk in chunks)
+
+        info.update({"features": list(chain.from_iterable(chunk_data))})
+        return cls(**info)
+
+
+class Raw:
+    def __init__(self, k=1, max_duplicates=None):
+        self.k = k
+        self.max_duplicates = max_duplicates if max_duplicates is not None else 5
+
+    def transform_image(self, X_image: ee.Image) -> ee.Image:
+        """No-op placeholder."""
+        return X_image
+
+    def transform_fc(self, fc: ee.FeatureCollection) -> ee.FeatureCollection:
+        """No-op placeholder."""
+        return fc
+
+    @property
+    def k_nearest(self):
+        """Return the number of requested nearest neighbors plus max_duplicates
+        in order to account for non-independent samples.
+        """
+        return self.k + self.max_duplicates
+
+    def get_ids(self, fc: FeatureCollection, id_field: str) -> list[Any]:
+        """Return the IDs from the feature collection."""
+        return list(map(int, fc.aggregate_array(id_field)))
+
+    def _set_classifier(
+        self,
+        *,
+        fc: ee.FeatureCollection,
+        class_property: ee.String,
+        input_properties: ee.List,
+    ) -> None:
+        """Set the EE minimumDistance classifier for this estimator."""
+        self.clf = ee.Classifier.minimumDistance(
```

Comment:
Yep, good idea.  I've done that in a recent commit and renamed `Raw._get_classifier` to `Raw._train_classifier` since that's it's only responsibility now.  Because I'm still calling it from both `Raw.train` and `Transformed.train` with different signatures, I think I still need this method.

Reply:
Ah, I missed that it was being used by both `train` methods, so keeping it in a separate `_train_classifier` method looks like a great solution!